### PR TITLE
Blaze: do not load the UI if the JSON API module is inactive

### PR DESCRIPTION
**Pull request by jeherve**
## Proposed changes:

The Blaze interface is currently not fully working when the JSON API module is inactive. Let's not promote Blaze in this case.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

* 1386-gh-dotcom-forge

## Does this pull request change what data or activity we track or use?

* No

## Testing instructions:

* Test on a WoA site
* Go to the Post list, in wp-admin: `wp-admin/edit.php?post_type=post`
* You should see the "Blaze" option when moving your mouse over a post row (if the post is published).
* Go to `https://yoursite.com/wp-admin/admin.php?page=jetpack_modules`
* Disable the JSON API module.
* Reload the post list in wp-admin.
* The Blaze option should not be there anymore.

### CODE REVIEW

**Review by kraftbj — DISMISSED** [review]
Approving this as a bandaid, but this suggests to me there is a JSON client that interacts with the site that isn't part of our WordPress.com group that skips the JSON API module check.

I left a comment with details on the original issue https://github.com/Automattic/dotcom-forge/issues/1386#issuecomment-1377959231

**Comment by kraftbj** [comment]
Marking as Author Reply to defer to @jeherve if he wants to explore a potential cause or to go with this PR. It is ready to merge pending his call.

**Comment by jeherve** [comment]
> Approving this as a bandaid, but this suggests to me there is a JSON client that interacts with the site that isn't part of our WordPress.com group that skips the JSON API module check.

Yep, this is really meant as a bandaid for now. The bug that requires this bandaid is being worked on right now.

File: projects/packages/blaze/src/class-blaze.php
Comment (dereksmart) [inline projects/packages/blaze/src/class-blaze.php]: How does this behave for WPCOM sites?

Comment (jeherve) [reply projects/packages/blaze/src/class-blaze.php]: Thanks for the reminder. I had assume it would always return true, but it turns out it doesn't behave like the old `Jetpack::is_module_active()` method.

I've added a wpcom check in 168dd5256f2cacaaa295b64ef3664ac9a10b53a3

**Review by dereksmart — APPROVED** [review]

**Pull request merged**
